test/transactionRunner.js

```diff
@@ -33,9 +32,6 @@ testing.runTests(function (testData, sst, cb) {
   try {
     var rawTx = ethUtil.toBuffer(testData.rlp)
     var tx = new Tx(rawTx)
-    if (testData.blocknumber !== String(common.homeSteadForkNumber.v)) {
```

Comment:
To keep this consistent with the changes to index.js, `if (testData.blocknumber < String(common.homeSteadForkNumber.v)) {` would be preferable here. However, as my PR to rewrite the `transactionRunner.js` file will be merged right after this PR, I think this change is okay.

Reply:
It's very likely that tests are failing here cause of the removal of this comparison and therefore have changed conditions.

I would just do a hard-coded comparison here and then instantiate the tx with the corresponding HF in the options, so:

```javascript
if (testData.blockNumber >== 1150000) {
  // instantiate a Homestead+ tx
} else {
  // instantiate a pre-homestead (chainstart) tx
}
```

Then we recreate the former conditions (you might even want to just use the same `!==` comparison) and it should be easier to get this passed for now. Dan can then do subsequent test optimization work in his own PR. Does this make sense?

Reply:
Excellent will give this a shot after work.